### PR TITLE
[Main Course] - Docker Volumes

### PR DESCRIPTION
**Pull request by Tina0330**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    作者：Tina，发布于2021-11-25，阅读时间：约6分钟，微信公众号文章链接：[:fontawesome-solid-link:]()
+
+## 1 前言
+在[Docker初体验](https://mp.weixin.qq.com/s/gfO5BiK9fqRtWf8rjP8mPA)一文中我们介绍了Docker的基本概念，之后我们又分享了如何[教你成功在Win10系统中运行docker](https://mp.weixin.qq.com/s/8B9ye55zpWCCVTA4g4fLQQ)和[使用GitHub Action自动构建和推送Docker镜像](https://mp.weixin.qq.com/s/BCzu346DvNga84vEUXUbTQ)，尝试过的朋友应该可以在Docker中为自己的程序代码创建镜像，运行容器，或是在`github action`中使用Docker等基本操作。今天，我们将说说如何Docker中持续化管理数据--数据卷(Volumes)的使用。
+
+## 2 管理数据的必要性
+初次体验Docker的人可能会发现，每当我们重启容器时，之前的数据都会丢失，又或是感觉数据很难转移。这些都是因为Docker的默认设置会将所有的文件都创建在一个容器中的某些可读容器层。换句话说，一个容器的数据是独立不能共享的，为了实现数据共享和数据迁移，就需要我们使用一些方法可持续化地管理数据。
+
+## 3 关于Volumes
+### 3.1 优势
```
Comment (Realvincentyuan) [inline docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md:17]: 最好空一行

File: docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+docker run -v /dbdata --name dbstore2 ubuntu /bin/bash
+```
+
+再从`backup.tar`文件中迁移到新的容器中：
+```shell
+ docker run --rm --volumes-from dbstore2 -v $(pwd):/backup ubuntu bash -c "cd /dbdata && tar xvf /backup/backup.tar --strip 1"
+```
+
+## 4 总结
+当部署服务时，功能的开发是一方面，数据的可持续化管理也是必不可少的一个部分。希望这篇文章可以帮助你在Docker学习过程中对Volumes有更深的理解并选择合适的命令管理容器中的数据。
```
Comment (jhqian0909) [inline docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md:118]: 最后一句话好像有点长

File: docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+<figure>
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/img/difference_ploy.png"  />
+  <figcaption>来自：Docker官方文档</figcaption>
+</figure>
+
+### 3.2 Volumes常见操作
+这里参考[官方文档](https://docs.docker.com/storage/volumes/ 'Docker Volumes')，我们介绍几种不同情况下对Volumes的使用。
+
+1.我们可通过以下基本命令创建及管理Volumes:
```
Comment (jhqian0909) [inline docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md:28]: 小标题后面可以空一格

**Review by Realvincentyuan — APPROVED** [review]
不同的样式间，最好空一行

File: docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md (hunk 1 above)
Comment (Tina0330) [reply docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md:18]: updated, thanks

File: docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md (hunk 2 above)
Comment (Tina0330) [reply docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md]: updated, thanks

File: docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+### 3.1 优势
+
+Volumes是一种由Docker创建并维护的数据管理机制。如下图所示，它最大的特点是将文件存放在Docker主机内，是不允许其他程序的访问，在不同容器间共享数据时具有较高的安全性；且可直接用Docker命令进行数据备份和数据迁移。除此之外，Volumes在Docker Desktop的表现性能比其他方式更加出色，这也是官方最推荐的方式。
+
+<figure>
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/img/difference_ploy.png"  />
+  <figcaption>来自：Docker官方文档</figcaption>
+</figure>
+
```
Comment (EchoTang04) [inline docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md:26]: 不知道是不是我眼花，感觉图有点糊。

**Review by EchoTang04 — APPROVED** [review]
LGTM, thanks Tina!

File: docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md (hunk 4 above)
Comment (Tina0330) [reply docs/Main_Course/Analytics_Tools/2021-11-25-Docker-Volumes.md:26]: 应该是截图放大了原因

**Pull request merged**